xonsh/main.py

```diff
@@ -224,8 +224,8 @@ def main(argv=None):
         # run a script contained in a file
         path = os.path.abspath(os.path.expanduser(args.file))
         if os.path.isfile(path):
-            sys.argv = args.args
-            env['ARGS'] = [args.file] + args.args
+            sys.argv = [args.file] + args.args
+            env['ARGS'] = sys.argv[:]  # $ARGS is not sys.argv
```

Comment:
My solution is also, well, more consistent: it always introduces bugs. :) Thus making harder to miss them. The Zen says about that, too.

> that is a python implementation detail

I suppose, "corner case" using of implementation details is a kind of covert bug.